arelle/FileSource.py

```diff
@@ -750,18 +895,21 @@ def gaeGet(key):
         return None
 
 
-def gaeDelete(key):
+def gaeDelete(key: str) -> bool:
+    assert gaeMemcache is not None
     chunk_keys = gaeMemcache.get(key)
     if chunk_keys is None:
         return False
-    if isinstance(chunk_keys, _STR_BASE):
+    if isinstance(chunk_keys, _STR_BASE): # type: ignore[name-defined]
         chunk_keys = []
     chunk_keys.append(key)
     gaeMemcache.delete_multi(chunk_keys)
     return True
 
 
-def gaeSet(key, bytesValue): # stores bytes, not string valye
+# Note 2022-09-12
+# Looks like maybe memcache of google.appengine.api is Python 2?
```

Comment:
As noted here, I believe this is legacy code but didn't want to introduce any breaking changes when adding type hints so I've let this remain here.